index.js

```diff
@@ -273,6 +296,7 @@ function processStyle(glStyle, map, baseUrl, host, path, accessToken) {
       setBackground(map, glLayer);
     } else {
       id = glLayer.source || getSourceIdByRef(glLayers, glLayer.ref);
+      // TODO: this technique assumes gl layers will be in particular order; fix
```

Comment:
My use case for this package will be fairly small mapbox styles, and I could easily see just editing them by hand.  I could see hours of pain for someone who didn't know that order mattered.

Reply:
Can you share an example style where this would fail? Still trying to understand.

Reply:
I think we can assume validated gl style documents. A validator is available: https://github.com/mapbox/mapbox-gl-js/tree/master/src/style-spec#gl-style-validate (not only as command line utility - it can also be used by an application).

Can this technique also be a problem with valid style documents?

Reply:
If there's a validator, then you're right; it probably doesn't matter.  Can I leave the comment in if I take off the "fix" part?  It took me awhile to figure out exactly how that worked, and if anyone else goes in there, it will probably save them some time...

Reply:
Ok.